anabot-1.sh

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (C) 2021  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+TESTTYPE="ui smoke anabot skip-on-rhel-8"
+
+. ${KSTESTDIR}/functions.sh
+
+kernel_args() {
+    local tmp_dir="${1}"
+
+    # Provide an installation source.
+    local repo_url="$( echo ${KSTEST_URL} | sed -e 's/^--url=//' )"
+
+    # Provide the anabot recipe.
+    cp ${KSTESTDIR}/${name}.xml ${tmp_dir}/http/example.xml
+    local httpd_url="$(cat ${tmp_dir}/httpd_url)"
+    local recipe_url="${httpd_url}/example.xml"
+
+    echo "${DEFAULT_BOOTOPTS} inst.graphical  anabot=${recipe_url} inst.repo=${repo_url}"
+}
+
+prepare_updates() {
+    local tmp_dir="${1}"
+    local updates_dir="${tmp_dir}/updates"
+    local updates_img="${tmp_dir}/updates.img"
+
+    local anabot_branch="main"
+    local anabot_url="https://pagure.io/anabot.git"
+    local anabot_img="${tmp_dir}/anabot.tar.gz"
+
+    # Use a fork.
+    # FIXME: Don't use the fork.
+    local anabot_url="https://pagure.io/forks/vponcova/anabot.git"
+    local anabot_branch="main-f35_support"
+
+    # Create an updates image with anabot.
+    # Use  --shallow-submodules.
+    (
+      cd "${tmp_dir}" && \
+      git clone --depth 1 --shallow-submodules -b "${anabot_branch}" "${anabot_url}" && \
+      cd anabot && \
+      git submodule --quiet update --init && \
+      ./make_updates.sh "${anabot_img}"
+    )
+
+    # Apply the anabot updates image.
+    apply_updates_image "file://${anabot_img}" "${updates_dir}"
+
+    # Apply the provided updates image if any.
+    apply_updates_image "${UPDATES}" "${updates_dir}"
+
+    # Create a new updates image.
+    create_updates_image "${updates_dir}" "${updates_img}"
+
+    # Provide the image. The function prints the URL.
+    upload_updates_image "${tmp_dir}" "${updates_img}"
+}
+
+inject_ks_to_initrd() {
+    echo "false"
+}
+
+validate() {
+    local tmp_dir="${1}"
+    copy_interesting_files_from_system "${tmp_dir}"
+    echo "SUCCESS" > "${tmp_dir}/RESULT"
+    check_result_file "${tmp_dir}"
```

Comment:
What is the benefit of this? Instead of creating the file and then checking it's creating, maybe we should just return 0? What are we checking here?

Reply:
The file might be needed for something else, outside of the test itself. I think it's good to move towards "always have the file".

I agree that return 0 would work just as well for the second line.

Reply:
Interesting, so basically we are solving required side effect here. Could you please add a comment if that is true ideally with explanation what will happen if the side effect is not fulfilled?

Reply:
The result file now contains the anabot errors if any.